client/state/ui/selectors/get-selected-site-slug.ts

```diff
@@ -8,7 +9,7 @@ import getSelectedSiteId from './get-selected-site-id';
  * @param  {object}  state Global state tree
  * @returns {?string}       Selected site slug
  */
-export default function getSelectedSiteSlug( state ) {
+export default function getSelectedSiteSlug( state: IAppState ) {
```

Comment:
This return type might be inferred by TS, but it may be worth making this explicit, since the JSDoc does not mention that it can return null and I don't think it can actually return undefined.

```suggestion
export default function getSelectedSiteSlug( state: IAppState ): null | string {
```